### DISCUSSION TITLE
Changing stream

### DISCUSSION THREAD

**Discussion by gerthalberg** [baldram/ESP_VS1053_Library#81] [Q&A]
Hi, I'm trying to change the the stream I'm listening to. I call stopSong and try a new connectToHost. In the headphones it appears, that the stream speeds up (to empty buffer?) I somehow always get a song stopped incorrectly and a panick when I connectToHost. Anyone got a suggestion as to what I'm doing wrong?

**Comment by Dr-Dawg** [comment]
As far as I remember, stopSong was used in the context of using VS10xx as a mp3 player (SD-Card, ..). I t sends fill bytes to your VS10xx device and tells it to cancel decoding the current file.

For changing the stream in the WebRadioDemo, changing the content of host, path and httpPort and calling client.stop(); should do the trick. In the main loop, the "if (!client.connected()) {..." part will connect to the new station.

Btw I don't think there's any way to change a stream exactly after a song, even looking for metadata won't work, because the radio stations do not send the data exactly between songs.
